airbyte-cdk/python/airbyte_cdk/test/mock_http/mocker.py

```diff
@@ -133,3 +133,7 @@ def wrapper(*args, **kwargs):  # type: ignore  # this is a very generic wrapper
                 return result
 
         return wrapper
+
+    def clear_all_matchers(self) -> None:
```

Comment:
When do we need that? I would assume that we instantiate one HttpMocker per test and hence we would not need to clear it. 

I might want to be opinionated on this because this would mean that a test has multiple reasons to fail and therefore the test could be improved/split

Reply:
This test https://github.com/airbytehq/airbyte/pull/48116/files#diff-0de03b21baa1dfcd1603abb7a1316defd40bc88b874b5861678b04960ee37b7bR217-R255
`test_given_known_error_when_read_brands_v3_report_then_skip_report`
fails because we try to introduce new mocker on line 237 and it fails because request was already mocked on previous cycle iteration (only response is changed) :
![image](https://github.com/user-attachments/assets/add96769-554b-4c5b-bd2e-202eeb7fc68d)

```
raise ValueError(f"Request {matcher.request} already mocked")
```


AFAIK, we cannot apply `pytest.mark.parametrize` for `unittest.Testcase`, so it is easier to clear the HttpMocker and declare it in every cycle inside the test

Reply:
I wouldn't mind having a private test method and a method per set of parameters. Something like:
```
class TestClass:
  def test_given_x_when_y_then_z(self):
    self._private_test_method(<...>)

  def _private_test_method(self, a, b, c): 
    <do something>
```

The reason I prefer this is that we can run them independently in IDEs like PyCharm (I don't know how to do that with `pytest.mark.parametrize` so if this is possible, please educate me). I also prefer the `setUp` and `tearDown` pattern from unittest but I wouldn't push for that too much as I understand it is subjective

Reply:
I get your point, we can split the tests, for sure, I can get to it later.

P.S. to run test from pytest.mark.parametrize you should run the whole test, and after that you'll be able to run specific case only
![image](https://github.com/user-attachments/assets/5d8dabac-8915-41d3-b8da-ebbf2c654267)

![image](https://github.com/user-attachments/assets/551a34d0-e071-4452-b02f-93960acd397d)

Reply:
This does not work for me. It creates this configuration (see image). However when I run it, it trigger the command `/Users/maxime/devel/code/airbyte/airbyte-cdk/python/.venv/bin/python /Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py --path /Users/maxime/devel/code/airbyte/airbyte-integrations/connectors/source-monday/unit_tests` which fails because of dependency issues. I don't know how PyCharm deduce this command from the run configuration as there is nothing about source-monday in the configuration...
![image](https://github.com/user-attachments/assets/da813982-f8b9-4f90-be7d-d789680ea5bd)